### PR TITLE
add module for reading and writing built by markers

### PR DESCRIPTION
**Pull request by wjwwood**
See: https://github.com/ros/catkin/issues/742

### CODE REVIEW

**Comment by tfoote** [comment]
+1

**Comment by jbohren** [comment]
+1

File: test/test_tool_detection.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+from catkin_pkg.tool_detection import get_previous_tool_used_on_the_space
+from catkin_pkg.tool_detection import mark_space_as_built_by
+
+from .util import in_temporary_directory
+
+
+@in_temporary_directory
+def test_get_previous_tool_used_on_the_space():
+    res = get_previous_tool_used_on_the_space('folder_that_does_not_exist')
+    assert(res is None, res)
```
Comment (dirk-thomas) [inline test/test_tool_detection.py]: `assert` is not a function but a statement. This check will always be true no matter what `res` is.

Same below.

Comment (wjwwood) [inline test/test_tool_detection.py]: Good catch, too much C lately...

https://github.com/ros-infrastructure/catkin_pkg/commit/e9b73902a1164ad7f0eb2bb16b358cef2f2300c7

**Comment by dirk-thomas** [comment]
lgtm

**Pull request merged**